From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: derverdox <lukasjonsson@outlook.de>
Date: Tue, 27 Jun 2023 12:57:46 +0200
Subject: [PATCH] Removing conversion from legacy components since it makes
 some items not stackable to other items that have lore / custom names.


diff --git a/src/main/java/net/minecraft/world/item/ItemStack.java b/src/main/java/net/minecraft/world/item/ItemStack.java
index d90cb7e28a8fbf8e8758817f49c36dd0539b2f62..f77c4e5732650d57fe9ceaf3cdfdb0e58ebe8142 100644
--- a/src/main/java/net/minecraft/world/item/ItemStack.java
+++ b/src/main/java/net/minecraft/world/item/ItemStack.java
@@ -207,7 +207,7 @@ public final class ItemStack {
         Component component = Component.Serializer.fromJson(json);
         if (component.getContents() instanceof net.minecraft.network.chat.contents.LiteralContents literalContents && literalContents.text().contains("\u00A7") && component.getSiblings().isEmpty()) {
             // Only convert if the root component is a single comp with legacy in it, don't convert already normal components
-            component = org.bukkit.craftbukkit.util.CraftChatMessage.fromString(literalContents.text())[0];
+            //component = org.bukkit.craftbukkit.util.CraftChatMessage.fromString(literalContents.text())[0];
         }
         return net.minecraft.nbt.StringTag.valueOf(org.bukkit.craftbukkit.util.CraftChatMessage.toJSON(component));
     }